src/components/payments/WestpacQuickStreamPaymentForm.tsx

```diff
@@ -0,0 +1,384 @@
+          } catch (error) {
+            console.warn('Failed to load script', error)
+            reject(error)
+          }
+        })
+
+        window.QuickstreamAPI.init({
+          publishableApiKey,
+        })
+        const trustedFrame = await new Promise<TrustedFrame>(
+          (resolve, reject) => {
+            window.QuickstreamAPI.creditCards.createTrustedFrame(
+              {
+                config: {
+                  supplierBusinessCode,
+                  threeDS2: true,
+                },
+                iframe: {
+                  width: '100%',
+                  height: '100%',
+                  style: {
+                    border: '1px solid #dedede',
+                    'border-radius': '2px',
+                    'margin-bottom': '0.75rem',
+                    'min-height': '400px',
+                    padding: '1.5rem',
+                    width: '100%',
+                    'background-color': 'white',
+                  },
+                },
+                showAcceptedCards: true,
+                showRequiredIndicators: true,
+              },
+              (errors, data) => {
+                if (errors) {
+                  reject(errors)
+                  return
+                }
+                resolve(data.trustedFrame)
+              },
+            )
+          },
+        )
+        const westpacQuickStreamForm = document.getElementById(
+          'westpac-quick-stream-form-id',
+        )
+        westpacQuickStreamForm?.addEventListener('submit', (event) => {
+          event.preventDefault()
+
+          setCompleteTransactionState({
+            isCompletingTransaction: true,
+            completeTransactionError: null,
+          })
+
+          trustedFrame.submitForm((errors, data) => {
+            if (errors) {
+              console.log('Invalid payment form submission', errors)
+              setCompleteTransactionState({
+                isCompletingTransaction: false,
+                completeTransactionError: null,
+              })
+              return
+            }
+
+            paymentService.westpacQuickStream
+              .completeTransaction({
+                singleUseTokenId: data.singleUseToken.singleUseTokenId,
+                formSubmissionResult,
+                formSubmissionPaymentId,
+                paymentSubmissionEvent,
+              })
+              .then(onCompleted)
+              .catch((error) => {
+                setCompleteTransactionState({
+                  isCompletingTransaction: false,
+                  completeTransactionError: error,
+                })
+              })
+          })
+        })
+
+        setLoadState({
+          isLoading: false,
+          loadError: null,
+        })
+      } catch (error) {
+        setLoadState({
+          isLoading: false,
+          loadError: error as Error,
+        })
+      }
+    }
+
+    run()
+
+    return () => {
+      document.body.appendChild(scriptEle)
```

Comment:
shouldn't this be cleaning up, like removing the event listeners etc

Reply:
I believe you are correct!!!